lib/IRGen/IRGen.cpp

```diff
@@ -169,6 +178,29 @@ void swift::performLLVMOptimizations(IRGenOptions &Opts, llvm::Module *Module,
     PMBuilder.addExtension(PassManagerBuilder::EP_EnabledOnOptLevel0,
                            addThreadSanitizerPass);
   }
+
+  if (Opts.SanitizeCoverage != SanitizerCoverageKind::None) {
+    PassManagerBuilder::ExtensionFn extensionFn = nullptr;
+    switch (Opts.SanitizeCoverage) {
+    case SanitizerCoverageKind::Function:
+      extensionFn =
+          addSanitizerCoveragePass<SanitizerCoverageOptions::SCK_Function>;
+      break;
+    case SanitizerCoverageKind::BB:
+      extensionFn = addSanitizerCoveragePass<SanitizerCoverageOptions::SCK_BB>;
+      break;
+    case SanitizerCoverageKind::Edge:
+      extensionFn =
+          addSanitizerCoveragePass<SanitizerCoverageOptions::SCK_Edge>;
+      break;
+    default:
```

Comment:
The switch is guarded by ``if (Opts.SanitizeCoverage != SanitizerCoverageKind::None)`` so adding a ``case SanitizerCoverageKind::None`` would not be useful because the case would not be reachable.

I could drop the guard add ``case SanitizerCoverageKind::None`` to do nothing in the switch statement but then I would need to guard the ``PMBuilder.addExtension(..)`` calls.